### DISCUSSION TITLE
DeepSource Static Analysis of the codebase

### DISCUSSION THREAD

**Discussion by de-sh** [linkerd/linkerd2#5250] [Show and tell] [answered]
Hey, I had recently gone through the [results of running the DeepSource static analyzer](https://deepsource.io/gh/de-sh/linkerd2/) on the linkerd2 codebase and there are various instances of potential bugs, low performance code and anti-patterns that might interest maintainers. 

I have [fixed a few of these issues](https://github.com/de-sh/linkerd2/commits?author=de-sh) and intend to open a PR, would it be a good idea to go ahead with this contribution?

Thanks!

**Comment by grampelberg** [comment, marked as answer]
We'd love help there!
